tests/test_endpoints.py

```diff
@@ -1,7 +1,156 @@
+        Schedule(
+            name="At 12:00 on day-of-month 20", minute="0", hour="12", day="20"
+        ),
+        date,
+    )
+    assert not test_schedule(
+        Schedule(name="Every minute past hour 13", hour="13"), date
+    )
+
+
+def init_schedule_on_db():
+    test_env = Environment(name="test_env")
+
+    admin_group = NotificationGroup(name="Admin")
+    admin_group.add_email("admin@some-test-server.somewhere")
+    producers_group = NotificationGroup(name="Producers")
+    producers_group.add_email("producer1@some-test-server.somewhere")
+    producers_group.add_email("producer2@some-test-server.somewhere")
+    bakery_group = NotificationGroup(name="Bakery")
+    bakery_group.add_email("bakery@some-test-server.somewhere")
+
+    # First job
+    baking_job = Job(name="Baking", type=JobType.SCHEDULED)
+    baking_context = Context(
+        environment=test_env,
+        job=baking_job,
+        package_name="baking",
+        package_version="1.0.1",
+    )
+
+    schedule_every_minute = Schedule(name="every minute")
+    baking_context.add_schedule(schedule_every_minute)
+
+    first_notification_baking = Notification(
+        on_retry=True, on_failure=True, job=baking_job, groups=[admin_group]
+    )
+    second_notification_baking = Notification(
+        on_retry=False,
+        on_failure=True,
+        job=baking_job,
+        groups=[producers_group, bakery_group],
+    )
+
+    # Second job
+    realtime_job = Job(name="Insert Realtime", type=JobType.TRIGGERED)
+    realtime_context = Context(
+        environment=test_env,
+        job=realtime_job,
+        package_name="insert_realtime",
+        package_version="2.1.0",
+    )
+
+    first_notification = Notification(
+        on_retry=True, on_failure=True, job=realtime_job, groups=[admin_group]
+    )
+
+    # Job that mustn't appear
+
+    stop_job = Job(name="Stop everything", type=JobType.SCHEDULED)
+    stop_context = Context(
+        environment=test_env,
+        job=stop_job,
+        package_name="shutdown",
+        package_version="1.0.0",
+    )
+
+    db.session.add_all(
+        [
+            test_env,
+            admin_group,
+            producers_group,
+            bakery_group,
+            baking_job,
+            baking_context,
+            schedule_every_minute,
+            first_notification_baking,
+            second_notification_baking,
+            realtime_job,
+            realtime_context,
+            first_notification,
+            stop_job,
+            stop_context,
+        ]
+    )
+    db.session.commit()
+
+
+def test_it_get_the_current_schedule(session, current_schedule):
+    init_schedule_on_db()
+    result = session.get("/".join([BASE_API, "schedule"]))
+    result_json = result.json()
+
+    actual_date = {"date": str(datetime.now())}
+    result_json.update(actual_date)
+    current_schedule.update(actual_date)
+
+    assert result.status_code == 200
+    assert result_json == current_schedule
```

Comment:
here you're testing that the full output of the endpoint matches the full saved output in the test data folder.

It's quite nice when you have 100% confidence that your output won't change, for instance if you are implementing a spec. 

Since we're far from that, by checking full input with full output, you're exposing yourself to broken tests whenever you'll find yourself adding/removing top keys in the input. 

Also, this is more like integration testing rather than unit-testing.

I'd recommend setting up smaller tests, that just check for the content of one or a couple of specific keys.

Reply:
Now I only test the presence of a single job